A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2322,54 +2329,20 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     getVLS() {
-        if (this.perfApprFlaps3) {
-            const dWeight = SimVar.GetSimVarValue("TOTAL WEIGHT", "kilograms") / 1000;
-            const cg = this.zeroFuelWeightMassCenter;
-            if (((isNaN(cg)) ? 24 : cg) < 25) {
-                switch (true) {
-                    case (dWeight <= 40): return 116;
-                    case (dWeight <= 45): return Math.ceil(116 + 0.4 * (dWeight - 40));
-                    case (dWeight <= 60): return Math.ceil(118 + 1.2 * (dWeight - 45));
-                    case (dWeight <= 75): return Math.ceil(136 + dWeight - 60);
-                    default: return Math.ceil(150 + dWeight - 75);
-                }
-            }
-            switch (true) {
-                case (dWeight <= 45): return 116;
-                case (dWeight <= 50): return Math.ceil(116 + 1.4 * (dWeight - 45));
-                case (dWeight <= 55): return Math.ceil(123 + 1.2 * (dWeight - 50));
-                case (dWeight <= 60): return Math.ceil(129 + dWeight - 55);
-                case (dWeight <= 65): return Math.ceil(134 + 1.2 * (dWeight - 60));
-                default: Math.ceil(140 + dWeight - 65);
-            }
-        } else {
-            const dWeight = SimVar.GetSimVarValue("TOTAL WEIGHT", "kilograms") / 1000;
-            const cg = this.zeroFuelWeightMassCenter;
-            if (((isNaN(cg)) ? 24 : cg) < 25) {
-                switch (true) {
-                    case (dWeight <= 50):
-                        return 116;
-                    case (dWeight >= 75):
-                        return Math.ceil(139 + .8 * (dWeight - 75));
-                    case (dWeight <= 55):
-                        return Math.ceil(116 + .8 * (dWeight - 50));
-                    case (dWeight <= 70):
-                        return Math.ceil(120 + dWeight - 55);
-                    default:
-                        return Math.ceil(135 + .8 * (dWeight - 70));
-                }
-            }
-            switch (true) {
-                case (dWeight <= 50):
-                    return 116;
-                case (dWeight >= 75):
-                    return Math.ceil(139 + .8 * (dWeight - 75));
-                case (dWeight <= 55):
-                    return Math.ceil(116 + .6 * (dWeight - 50));
-                default:
-                    return Math.ceil(119 + dWeight - 55);
-            }
-        }
+        return SimVar.GetSimVarValue("L:A32NX_SPEEDS_VLS", "number");
+    }
+
+    /**
+     * VLS for landing config currently selected in the MCDU (CONF3 or FULL)
+     */
+    getPerfApprVLS() {
+        // TODO should be based on predicted landing weight, not current weight
+        const dWeight = SimVar.GetSimVarValue("TOTAL WEIGHT", "kilograms") / 1000;
+        const flaps = this.perfApprFlaps3 ? 3 : 4;
+        const gear = true;
+        const takeoff = false;
+        const speeds = new NXSpeeds(dWeight, flaps, gear, takeoff);
+        return speeds.vls;
```

Comment:
I suggest adding an explicit/new entry to the nxspeeds function to explicitly calculate and return the vls